geth/mailservice/mailservice_api.go

```diff
@@ -70,38 +150,65 @@ func (api *PublicAPI) RequestMessages(_ context.Context, r MessagesRequest) (boo
 
 	setMessagesRequestDefaults(&r)
 
-	shh, err := api.provider.WhisperService()
+	shh, err := api.provider().WhisperService()
 	if err != nil {
 		return false, err
 	}
-
-	node, err := api.provider.Node()
+	server, err := api.provider().Server()
 	if err != nil {
 		return false, err
 	}
-
-	mailServerNode, err := discover.ParseNode(r.MailServerPeer)
+	peer, err := api.choosePeer()
 	if err != nil {
-		return false, fmt.Errorf("%v: %v", ErrInvalidMailServerPeer, err)
+		return false, err
 	}
-
+	// renew gc timer
+	api.newConnectedPeers <- peer
```

Comment:
Between `peer, err := api.choosePeer()` and `api.newConnectedPeers <- peer` the peer can be removed due to  timeout.